src/utils/view-transition.ts

```diff
@@ -23,3 +23,8 @@ export async function startViewTransition(
     if (finishedCallback) Promise.resolve(finishedCallback());
   });
 }
+
+export function resolvePageTransitionPromise() {
+  pageTranstionResolver$.value?.("");
+  pageTranstionResolver$.value = void 0;
```

Comment:
what does these two lines do?

Reply:
Consider it as a trick to make the page transition work. The transition needs to know the view before change and the view after change. The view before change is captured when a button is clicked, and we use a promise that resolves in the view components' effect call. Basically, it is a `resolve` function from the view transition handler, then we call it when the new view is loaded. Previously, this appeared in every view component.